main/middleware/apisix_user_test.py

```diff
@@ -62,6 +62,45 @@ def test_get_request(mocker, mock_login):
     assert user.global_id == apisix_user_info["sub"]
 
 
+@pytest.mark.django_db(transaction=True)
+def test_get_request_existing_user_no_globalid(mocker, mock_login):
+    """Test that a valid request creates a new user."""
+    close_old_connections()
+    user = UserFactory.create(email=apisix_user_info["email"], global_id=None)
+    mock_request = mocker.Mock(
+        META={
+            "HTTP_X_USERINFO": b64encode(json.dumps(apisix_user_info).encode()),
+        },
+        user=AnonymousUser(),
+    )
+    apisix_middleware = ApisixUserMiddleware(mocker.Mock())
+    apisix_middleware.process_request(mock_request)
+    mock_login.assert_called_once()
+    user.refresh_from_db()
+    assert user.username == user.username
+    assert user.email == apisix_user_info["email"]
+    assert user.global_id == apisix_user_info["sub"]
+
+
+@pytest.mark.django_db(transaction=True)
+def test_get_request_existing_user_with_globalid(mocker, mock_login):
+    """Test that a valid request creates a new user."""
+    close_old_connections()
+    user = UserFactory.create(email=apisix_user_info["email"], global_id="abc123")
```

Comment:
I'm not sure I follow this test because I don't think this will match on the `filter()` because it doesn't match on `global_id` or null `global_id` + `email`. I would've expected this to fail because it'll try to create a user and conflict.

Reply:
Probably worth having this test, but it should wrap `process_request()` in a `pytest.raises(IntegrityError)` and then you don't need any other asserts.-